yt_dlp/extractor/qqmusic.py

```diff
@@ -1,245 +1,329 @@
+            'id': '000Q00f213YzNV',
+            'title': '桃几OvO',
+            'description': '小破站小唱见~希望大家喜欢听我唱歌~！',
+            'thumbnail': r're:^https?://.*\.jpg(?:$|[#?])',
+        },
+        'playlist_count': 12,
+        'playlist': [{
+            'info_dict': {
+                'id': '0016cvsy02mmCl',
+                'ext': 'mp3',
+                'title': '群青',
+                'album': '桃几2021年翻唱集',
+                'release_date': '20210913',
+                'duration': 248,
+                'creators': ['桃几OvO'],
+                'genres': ['Pop'],
+                'description': 'md5:4296005a04edcb5cdbe0889d5055a7ae',
+                'size': 3970822,
+                'thumbnail': r're:^https?://.*\.jpg(?:$|[#?])',
+            },
+        }],
+    }]
+
+    _PAGE_SIZE = 50
+
+    def _fetch_page(self, mid, page_size, page_num):
+        data = self.make_fcu_req({'req_1': {
+            'module': 'music.web_singer_info_svr',
+            'method': 'get_singer_detail_info',
+            'param': {
+                'sort': 5,
+                'singermid': mid,
+                'sin': page_num * page_size,
+                'num': page_size,
+            }}}, mid, note=f'Downloading page {page_num}')
+        yield from traverse_obj(data, ('req_1', 'data', 'songlist', ..., {lambda x: self.url_result(
+            f'https://y.qq.com/n/ryqq/songDetail/{x["mid"]}', QQMusicIE, x['mid'], x.get('title'))}))
 
     def _real_extract(self, url):
         mid = self._match_id(url)
+        init_data = self.download_init_data(url, mid)
 
-        entries = self.get_entries_from_page(mid)
-        singer_page = self._download_webpage(url, mid, 'Download singer page')
-        singer_name = self._html_search_regex(
-            r"singername\s*:\s*'(.*?)'", singer_page, 'singer name', default=None)
-        singer_desc = None
-
-        if mid:
-            singer_desc_page = self._download_xml(
-                'http://s.plcloud.music.qq.com/fcgi-bin/fcg_get_singer_desc.fcg', mid,
-                'Donwload singer description XML',
-                query={'utf8': 1, 'outCharset': 'utf-8', 'format': 'xml', 'singermid': mid},
-                headers={'Referer': 'https://y.qq.com/n/yqq/singer/'})
+        return self.playlist_result(
+            OnDemandPagedList(functools.partial(self._fetch_page, mid, self._PAGE_SIZE), self._PAGE_SIZE),
+            mid, **traverse_obj(init_data, ('singerDetail', {
+                'title': ('basic_info', 'name', {str}),
+                'description': ('ex_info', 'desc', {str}),
+                'thumbnail': ('pic', 'pic', {url_or_none}),
+            })))
 
-            singer_desc = singer_desc_page.find('./data/info/desc').text
 
-        return self.playlist_result(entries, mid, singer_name, singer_desc)
+class QQPlaylistBaseIE(InfoExtractor):
+    def _extract_entries(self, info_json, path):
+        for song in traverse_obj(info_json, path):
+            song_mid = song['songmid']
```

Comment:
This should not be fatal. I would have liked to include the filter in the `traverse_obj` call, but I don't see an immediate way how.
Unless you can think of something better, I would do:
```suggestion
            song_mid = song.get('songmid')
            if not song_mid:
                continue
```

Reply:
I think we could do something like `for song in traverse_obj(info_json, (*path, all, lambda _, v: v['songmid'])):`, but I am not sure if we want this to be fatal or not. @bashonly

Reply:
If a single song from a playlist is missing the `songmid` for whatever reason, we don't want the entire playlist extraction to fail.

Reply:
I would write it as non-fatal. Both the traversal filtering and sepro's suggestion would work, but I think the `if not ....: continue` pattern is easier to read